pkg/osquery/table/program_icons_windows.go

```diff
@@ -0,0 +1,195 @@
+package table
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"hash/crc64"
+	"image/png"
+	"os"
+
+	"strings"
+
+	"github.com/kolide/osquery-go/plugin/table"
+	"github.com/nfnt/resize"
+	"golang.org/x/sys/windows/registry"
+
+	"github.com/mat/besticon/ico"
+)
+
+var crcTable = crc64.MakeTable(crc64.ECMA)
+
+type icon struct {
+	base64 string
+	hash   uint64
+}
+
+func ProgramIcons() *table.Plugin {
+	columns := []table.ColumnDefinition{
+		table.TextColumn("name"),
+		table.TextColumn("version"),
+		table.TextColumn("icon"),
+		table.TextColumn("hash"),
+	}
+	return table.NewPlugin("kolide_program_icons", columns, generateProgramIcons)
+}
+
+func generateProgramIcons(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	var results []map[string]string
+
+	results = append(results, generateUninstallerProgramIcons()...)
+	results = append(results, generateInstallersProgramIcons()...)
+
+	return results, nil
+}
+
+func generateUninstallerProgramIcons() []map[string]string {
+	var uninstallerIcons []map[string]string
+
+	uninstallRegPaths := map[registry.Key][]string{
+		registry.LOCAL_MACHINE: append(expandRegistryKey(registry.LOCAL_MACHINE, `SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\*`),
+			expandRegistryKey(registry.LOCAL_MACHINE, `\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall\*`)...),
+		registry.USERS: expandRegistryKey(registry.USERS, `*\Software\Microsoft\Windows\CurrentVersion\Uninstall\*`),
+	}
+
+	for key, paths := range uninstallRegPaths {
+		for _, path := range paths {
+			key, err := registry.OpenKey(key, path, registry.ALL_ACCESS)
```

Comment:
```suggestion
			key, err := registry.OpenKey(key, path, registry. READ)
```

do we need that much permission?